src/cpp/session/modules/SessionRCompletions.R

```diff
@@ -2028,6 +2028,31 @@ assign(x = ".rs.acCompletionTypes",
    excludeArgsFromObject <- .rs.setEncodingUnknownToUTF8(excludeArgsFromObject)
    filePath <- .rs.setEncodingUnknownToUTF8(filePath)
    
+   # if base::readline() is on the stack, try to extract choices i.e. (yes/no)
+   # and offer those as completions.
+   nframes <- sys.nframe()
+   for (i in seq_along(nframes)) {
+      fun <- sys.function(i)
+      if (identical(fun, base::readline))
+      {
+         frame <- sys.frame(i)
+         prompt <- frame$prompt
```

Comment:
We should double-check that prompt is character here, since R does accept things like `readline(quote(wat))`.